core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -67,9 +67,7 @@ object StaticFile {
 
   def fromFile(f: File, start: Long, end: Long, buffsize: Int, req: Option[Request])
                         (implicit es: ExecutorService): Option[Response] = {
-
-    if (start < 0 || end < start || buffsize <= 0)
-      throw new Exception(s"start: $start, end: $end, buffsize: $buffsize")
+    require (start >= 0 && end > start && buffsize > 0, s"start: $start, end: $end, buffsize: $buffsize")
```

Comment:
One thing to consider is making the return types some form of ```Task```. That will put the error handling in line with the rest of the library. It makes me sad to make things asynchronous when they need not be, but I think thats justified considering the pervasiveness of ```Task``` in the library and sparsity of exceptions. I think that might be a separate PR if we go that route.